hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/utils/FileServiceUtils.java

```diff
@@ -92,4 +112,73 @@ public static void validateAndAddRequiredKeys(
 
         if (listKeys != null && listKeys.hasKeys()) for (Key key : listKeys.keys()) context.requireKey(key);
     }
+
+    /**
+     * This version of the method requires that the file exist in state and not be deleted. If the
+     * file is empty this call will fail with {@link ResponseCodeEnum#INVALID_FILE_ID}. If the
+     * special file id is greater than special file config , this call will fail with {@link
+     * ResponseCodeEnum#INVALID_FILE_ID}. If the file has no keys or empty list, this call will fail
+     * with {@link ResponseCodeEnum#UNAUTHORIZED}. If the file is already deleted, this call will
+     * fail with {@link ResponseCodeEnum#FILE_DELETED}.
+     *
+     * @param handleContext the handle context for the transaction.
+     * @param fileStore the file store to fetch the metadata of specified file id
+     * @param fileId the file id to validate and to fetch the metadata
+     * @param canBeDeleted flag indicating that the file may be deleted, and the call should not
```

Comment:
I do not understand that parameter

Reply:
I changed a little bit the description but it is needed for the file system to undelete the file because the flag is deleted but we need to undelete it and when it is verifying the file it will crash it.